### PR TITLE
[flang] Add translateNameToFrontendMangledName fct

### PR DESCRIPTION
**Pull request by clementval**
Sync back after upstreaming

### CODE REVIEW

**Review by kiranchandramohan — APPROVED** [review]
LGTM.

File: flang/include/flang/Optimizer/Dialect/FIRTypes.td [hunk 1]
```diff
@@ -324,6 +324,8 @@ def fir_RecordType : FIR_Type<"Record", "type"> {
     void finalize(llvm::ArrayRef<TypePair> lenPList,
                   llvm::ArrayRef<TypePair> typeList);
 
+    std::string getLoweredName() const;
```
Comment (schweitzpgi) [inline flang/include/flang/Optimizer/Dialect/FIRTypes.td:327]: This looks fine other than I don't know what a "lowered name" actually is. What is meant by here (why this is useful) needs to be documented here in the td file.

Comment (clementval) [reply flang/include/flang/Optimizer/Dialect/FIRTypes.td:327]: I actually don't know what this string is supposed to mean and this is the only name I came up with during upstreaming. Do you have any suggestion for a better name?

Comment (schweitzpgi) [reply flang/include/flang/Optimizer/Dialect/FIRTypes.td:327]: It looks to me like it is translating a FIR uniqued name to the mangled name produced by the front end's own distinct mangling scheme.

**Review by schweitzpgi — APPROVED** [review]
Thanks, Valentin.

**Pull request merged**